amqp_log_writer.go

```diff
@@ -94,7 +95,9 @@ func (w *amqpLogWriter) Write(p []byte) (int, error) {
 		if err != nil {
 			logger.WithField("err", err).Error("couldn't write 'log length exceeded' error message to log")
 		}
-		return 0, ErrWrotePastMaxLogLength
+		gocontext.Canceled = ErrWrotePastMaxLogLength
```

Comment:
This is probably going to break other things, since it replaces `Canceled` globally -- I believe _every_ canceled context will now be `ErrWrotePastMaxLogLength `. Not sure if there is any other way of setting a custom cancel error though.

Reply:
That's a good point, I'm hoping https://github.com/travis-ci/worker/pull/492/commits/dee958c71883fcef51e9cfa00cea076e92484a85 should fix it.